provision/provision.go

```diff
@@ -0,0 +1,162 @@
+// This package will fetch the recommendation from the recommendation Queue and provision the scale in/out
+// based on command.
+package provision
+
+import (
+	"scaling_manager/cluster"
+	"time"
+)
+
+var counter uint8 = 1
+
+// This struct contains the State of the opensearch scaling manager
+// States can be following:
+//  1. provision
+//  2. provisioning
+//  3. provisioning_completed
+//  4. provisioning_failed
+//  5. provisioned_successfully
+type State struct {
+	// CurrentState indicate the current state of the scaling manager
+	CurrentState string
+	// PreviousState indicates the previous state of the scaling manager
+	PreviousState string
+	// Remark indicates the additional remarks for the state of the scaling manager
+	Remark string
```

Comment:
Add a field called timestamp which will have the timestamp when state was provisioned_successfully